website/server/models/group.js

```diff
@@ -786,16 +786,15 @@ function _getUserUpdateForQuestReward (itemToAward, allAwardedItems) {
 
 async function _updateUserWithRetries (userId, updates, numTry = 1, query = {}) {
   query._id = userId;
-  return await User.update(query, updates).exec()
-    .then((raw) => {
-      return raw;
-    }).catch((err) => {
-      if (numTry < MAX_UPDATE_RETRIES) {
-        return _updateUserWithRetries(userId, updates, ++numTry, query);
-      } else {
-        throw err;
-      }
-    });
+  try {
+    return await User.update(query, updates).exec();
+  } catch (err) {
+    if (numTry < MAX_UPDATE_RETRIES) {
+      return _updateUserWithRetries(userId, updates, ++numTry);
```

Comment:
shouldn't query be passed here? Otherwise query will be {} at the second retry an update the user even if it doesn't match anymore?

Reply:
Good catch, I totally missed that. I’ll fix it up